doc/changelog/03-notations/12984-master+import-notation-make-active-again.rst

```diff
@@ -0,0 +1,9 @@
+- **Changed:**
+  Redeclaring a notation reactivates also its printing rule; in
+  particular a second :cmd:`Import` of the same module reactivates the
+  printing rules declared in this module; this is a rare but possible
+  source of incompatibilities
```

Comment:
I would not call this a "source of incompatibilities" if they can only be found in output tests.

Reply:
It is a change in behavior, for sure. It can break how goals are displayed in people's proofs. Sure, the old scripts will still work, but working *on* them could well be severely impacted by all notation being broken.

Reply:
So, "... this is a rare but possible source of changes of behavior"?

Reply:
What about being quite explicit and writing "this can lead to a change in printing behavior if different notations are selected"?

Reply:
Finally, I removed the "this is a rare but possible source of incompatibilities" since there are no apparent incompatibilities when combined with #12986. I hope this is ok for the both of you.

Reply:
Sorting by priority however is its own incompatibility, compared to "first import wins" before.

It is true that there is no known incompatibility from this, but that is because most of the CI doesn't do any printing testing. So there not being any known breakage reflects a lack of test coverage more than anything else, I think.

So given all that, I think the incompatibility still should be clearly called out, so that people know what to look for when they are affected.

Reply:
@Zimmi48, @RalfJung, what about the following then:
```
Redeclaring a notation reactivates also its printing rule; in 
particular a second :cmd:`Import` of the same module reactivates the 
printing rules declared in this module; this leads in theory to  
changes of behavior in printing though this seems to be mitigated in
general by the adoption in `#12986 
<https://github.com/coq/coq/pull/12986>`_ of a priority given to     
notations which match a larger part of the term to print
```

Reply:
LGTM